analysis/processors/bbww.py

```diff
@@ -510,29 +518,35 @@ def process_shift(self, events, shift_name):
         ###
         # Calculate derivatives
         ###
+        mbb = np.zeros(len(events), dtype="float")
+        mqq = np.zeros(len(events), dtype="float")
+        q2pt = np.zeros(len(events), dtype="float")
+
+
+        j_candidates = j_soft[ak.argsort(j_soft.particleNetAK4_QvsG, axis=1, ascending=False)] #particleNetAK4_QvsG btagPNetQvG
 
-        j_candidates = j_soft[ak.argsort(j_soft.particleNetAK4_QvsG, axis=1, ascending=False)]#particleNetAK4_QvsG btagPNetQvG
         j_candidates = j_candidates[:, :4] #consider only the first 4
+
         j_candidates = j_candidates[ak.argsort(j_candidates.particleNetAK4_B, axis=1, ascending=False)]#particleNetAK4_B btagPNetB
 
-        try:
-            bb = j_candidates[:, 0] + j_candidates[:, 1]
-            mbb = bb.mass
-        except:
-            mbb = np.zeros(len(events), dtype='float')
 
-        try:
-            qq = j_candidates[:, -1] + j_candidates[:, -2]
-            mqq = qq.mass
-        except:
-            mqq = np.zeros(len(events), dtype='float')
-        
+        valid_jets = ak.num(j_candidates) >= 2
+
+        if ak.any(valid_jets):  # Proceed only if there are valid events
```

Comment:
You technically do not need an "if" statement here. That's kind of the beauty of columnar analysis :)